project/SparkBuild.scala

```diff
@@ -358,7 +358,7 @@ object SparkBuild extends PomBuild {
       "-doc-title", "Spark " + version.value.replaceAll("-SNAPSHOT", "") + " ScalaDoc"
     ) ++ {
       // Do not attempt to scaladoc javadoc comments under 2.12 since it can't handle inner classes
-      if (scalaBinaryVersion.value == "2.12") Seq("-no-java-comments") else Seq.empty
+      if (scalaBinaryVersion.value == "2.13") Seq("-no-java-comments") else Seq.empty
```

Comment:
@srowen @easel I've found a leftover Scala 2.12 version check here, but when using the check method provided in https://github.com/apache/spark/pull/20042: `sbt tags/publishLocal`, I couldn't distinguish the scala doc built with or without `-no-java-comments`. I hope to get your help,  how should we determine whether to keep the option `-no-java-comments` for Scala 2.13? Thanks ~

Reply:
I think you should just remove this part, right? does not apply in Scala 2.13

Reply:
Thanks for your help @srowen , but I found that there are errors when executing `build/sbt doc -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive -Pvolcano`. Let me fix them first.